File: Emplyfi/ems-backend/src/main/java/com/ems/ems_backend/service/EmployeeService.java
package com.ems.ems_backend.service;

import java.util.List;

import com.ems.ems_backend.dto.EmployeeDto;

public interface EmployeeService {

	// Create
	EmployeeDto createEmployee(EmployeeDto employeeDto);

	// Get employee
	EmployeeDto getEmployeeById(Long employeeId);

	// Get all employees
	List<EmployeeDto> getAllEmployees();

	// Update employee
	EmployeeDto updateEmployee(Long employeeId, EmployeeDto updateEmployee);
	
	// Delete Employee
	void deleteEmployee(Long employeeId);
	
}

File: Emplyfi/ems-frontend/src/services/EmployeeService.js
import axios from "axios";

const BASE_URL = 'http://localhost:8080/employees';

export const listEmployees = () => axios.get(`${BASE_URL}/getall`);

export const createEmployee = (employee) => axios.post(`${BASE_URL}/create`, employee);

export const getEmployee = (employeeId) => axios.get(`${BASE_URL}/${employeeId}`);

export const updateEmployee = (employeeId, employee) => axios.put(`${BASE_URL}/updateEmployee/${employeeId}`, employee);

export const deleteEmployee = (employeeId) => axios.delete(`${BASE_URL}/deleteEmployee/${employeeId}`);